File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/widgetassertions/ComboBoxes_ViewModel.cs
/// <filename>
///     ComboBoxes_ViewModel.cs
/// </filename>
using System.Collections.Generic;
using widgetassertions;

namespace widgetassertions
{
    public class ComboBoxes_ViewModel
    {
        private System.Collections.Generic.List<string> MyOptionsComboBoxEntries = new System.Collections.Generic.List<string>();

        public virtual System.Collections.Generic.List<string> getMyOptionsComboBoxEntries()
        {
            return this.MyOptionsComboBoxEntries;
        }

        private string MyOptionsComboBoxSelectedEntry;

        public virtual string getMyOptionsComboBoxSelectedEntry()
        {
            return this.MyOptionsComboBoxSelectedEntry;
        }

        private bool isMyOptionsComboBoxVisible;

        public virtual bool getIsMyOptionsComboBoxVisible()
        {
            return this.isMyOptionsComboBoxVisible;
        }

        private bool isMyOptionsComboBoxEnabled;

        public virtual bool getIsMyOptionsComboBoxEnabled()
        {
            return this.isMyOptionsComboBoxEnabled;
        }

        private System.Collections.Generic.List<string> MyOptionsWithFreeTextComboBoxEntries = new System.Collections.Generic.List<string>();

        public virtual System.Collections.Generic.List<string> getMyOptionsWithFreeTextComboBoxEntries()
        {
            return this.MyOptionsWithFreeTextComboBoxEntries;
        }

        private string MyOptionsWithFreeTextComboBoxSelectedEntry;

        public virtual string getMyOptionsWithFreeTextComboBoxSelectedEntry()
        {
            return this.MyOptionsWithFreeTextComboBoxSelectedEntry;
        }

        private string MyOptionsWithFreeTextComboBoxText;

        public virtual string getMyOptionsWithFreeTextComboBoxText()
        {
            return this.MyOptionsWithFreeTextComboBoxText;
        }

        private bool isMyOptionsWithFreeTextComboBoxVisible;

        public virtual bool getIsMyOptionsWithFreeTextComboBoxVisible()
        {
            return this.isMyOptionsWithFreeTextComboBoxVisible;
        }

        private bool isMyOptionsWithFreeTextComboBoxEnabled;

        public virtual bool getIsMyOptionsWithFreeTextComboBoxEnabled()
        {
            return this.isMyOptionsWithFreeTextComboBoxEnabled;
        }

        private System.Collections.Generic.List<widgetassertions.ComboBoxes_ViewModelMyOptionsWithFixedChoicesOption> MyOptionsWithFixedChoicesComboBoxEntries = new System.Collections.Generic.List<widgetassertions.ComboBoxes_ViewModelMyOptionsWithFixedChoicesOption>();

        public virtual System.Collections.Generic.List<widgetassertions.ComboBoxes_ViewModelMyOptionsWithFixedChoicesOption> getMyOptionsWithFixedChoicesComboBoxEntries()
        {
            return this.MyOptionsWithFixedChoicesComboBoxEntries;
        }

        private widgetassertions.ComboBoxes_ViewModelMyOptionsWithFixedChoicesOption MyOptionsWithFixedChoicesComboBoxSelectedEntry;

        public virtual widgetassertions.ComboBoxes_ViewModelMyOptionsWithFixedChoicesOption getMyOptionsWithFixedChoicesComboBoxSelectedEntry()
        {
            return this.MyOptionsWithFixedChoicesComboBoxSelectedEntry;
        }

        private System.Collections.Generic.List<widgetassertions.ComboBoxes_ViewModelMyOptionsWithDisplayValueEntry> MyOptionsWithDisplayValueComboBoxEntries = new System.Collections.Generic.List<widgetassertions.ComboBoxes_ViewModelMyOptionsWithDisplayValueEntry>();

        public virtual System.Collections.Generic.List<widgetassertions.ComboBoxes_ViewModelMyOptionsWithDisplayValueEntry> getMyOptionsWithDisplayValueComboBoxEntries()
        {
            return this.MyOptionsWithDisplayValueComboBoxEntries;
        }

        private string MyOptionsWithDisplayValueComboBoxSelectedEntry;

        public virtual string getMyOptionsWithDisplayValueComboBoxSelectedEntry()
        {
            return this.MyOptionsWithDisplayValueComboBoxSelectedEntry;
        }

        private System.Collections.Generic.List<widgetassertions.ComboBoxes_ViewModelMyOptionsWithFixedChoicesAndDisplayValueEntry> MyOptionsWithFixedChoicesAndDisplayValueComboBoxEntries = new System.Collections.Generic.List<widgetassertions.ComboBoxes_ViewModelMyOptionsWithFixedChoicesAndDisplayValueEntry>();

        public virtual System.Collections.Generic.List<widgetassertions.ComboBoxes_ViewModelMyOptionsWithFixedChoicesAndDisplayValueEntry> getMyOptionsWithFixedChoicesAndDisplayValueComboBoxEntries()
        {
            return this.MyOptionsWithFixedChoicesAndDisplayValueComboBoxEntries;
        }

        private widgetassertions.ComboBoxes_ViewModelMyOptionsWithFixedChoicesAndDisplayValueOption MyOptionsWithFixedChoicesAndDisplayValueComboBoxSelectedEntry;

        public virtual widgetassertions.ComboBoxes_ViewModelMyOptionsWithFixedChoicesAndDisplayValueOption getMyOptionsWithFixedChoicesAndDisplayValueComboBoxSelectedEntry()
        {
            return this.MyOptionsWithFixedChoicesAndDisplayValueComboBoxSelectedEntry;
        }

        public virtual void myOptionsWithFixedChoicesComboBoxEntrySelected(string entryName)
        {
        }

        public virtual void myOptionsWithDisplayValueComboBoxEntrySelected(string entryName)
        {
        }

        public virtual void myOptionsWithFixedChoicesAndDisplayValueComboBoxEntrySelected(string entryName)
        {
        }
    }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/widgetassertions/ComboBoxes_ViewModel.cpp
#include "ComboBoxes_ViewModel.h"
#include <vector>
#include "ComboBoxes_ViewModelMyOptionsWithFixedChoicesOption.h"
#include "ComboBoxes_ViewModelMyOptionsWithDisplayValueEntry.h"
#include "ComboBoxes_ViewModelMyOptionsWithFixedChoicesAndDisplayValueEntry.h"
#include "ComboBoxes_ViewModelMyOptionsWithFixedChoicesAndDisplayValueOption.h"
#include <string>

namespace widgetassertions
{
  std::vector<std::string> ComboBoxes_ViewModel::getMyOptionsComboBoxEntries()
  {
    return this->MyOptionsComboBoxEntries;
  }
  std::string ComboBoxes_ViewModel::getMyOptionsComboBoxSelectedEntry()
  {
    return this->MyOptionsComboBoxSelectedEntry;
  }
  bool ComboBoxes_ViewModel::getIsMyOptionsComboBoxVisible()
  {
    return this->isMyOptionsComboBoxVisible;
  }
  bool ComboBoxes_ViewModel::getIsMyOptionsComboBoxEnabled()
  {
    return this->isMyOptionsComboBoxEnabled;
  }
  std::vector<std::string> ComboBoxes_ViewModel::getMyOptionsWithFreeTextComboBoxEntries()
  {
    return this->MyOptionsWithFreeTextComboBoxEntries;
  }
  std::string ComboBoxes_ViewModel::getMyOptionsWithFreeTextComboBoxSelectedEntry()
  {
    return this->MyOptionsWithFreeTextComboBoxSelectedEntry;
  }
  std::string ComboBoxes_ViewModel::getMyOptionsWithFreeTextComboBoxText()
  {
    return this->MyOptionsWithFreeTextComboBoxText;
  }
  bool ComboBoxes_ViewModel::getIsMyOptionsWithFreeTextComboBoxVisible()
  {
    return this->isMyOptionsWithFreeTextComboBoxVisible;
  }
  bool ComboBoxes_ViewModel::getIsMyOptionsWithFreeTextComboBoxEnabled()
  {
    return this->isMyOptionsWithFreeTextComboBoxEnabled;
  }
  std::vector<ComboBoxes_ViewModelMyOptionsWithFixedChoicesOption> ComboBoxes_ViewModel::getMyOptionsWithFixedChoicesComboBoxEntries()
  {
    return this->MyOptionsWithFixedChoicesComboBoxEntries;
  }
  ComboBoxes_ViewModelMyOptionsWithFixedChoicesOption ComboBoxes_ViewModel::getMyOptionsWithFixedChoicesComboBoxSelectedEntry()
  {
    return this->MyOptionsWithFixedChoicesComboBoxSelectedEntry;
  }
  std::vector<ComboBoxes_ViewModelMyOptionsWithDisplayValueEntry> ComboBoxes_ViewModel::getMyOptionsWithDisplayValueComboBoxEntries()
  {
    return this->MyOptionsWithDisplayValueComboBoxEntries;
  }
  std::string ComboBoxes_ViewModel::getMyOptionsWithDisplayValueComboBoxSelectedEntry()
  {
    return this->MyOptionsWithDisplayValueComboBoxSelectedEntry;
  }
  std::vector<ComboBoxes_ViewModelMyOptionsWithFixedChoicesAndDisplayValueEntry> ComboBoxes_ViewModel::getMyOptionsWithFixedChoicesAndDisplayValueComboBoxEntries()
  {
    return this->MyOptionsWithFixedChoicesAndDisplayValueComboBoxEntries;
  }
  ComboBoxes_ViewModelMyOptionsWithFixedChoicesAndDisplayValueOption ComboBoxes_ViewModel::getMyOptionsWithFixedChoicesAndDisplayValueComboBoxSelectedEntry()
  {
    return this->MyOptionsWithFixedChoicesAndDisplayValueComboBoxSelectedEntry;
  }
  void ComboBoxes_ViewModel::myOptionsWithFixedChoicesComboBoxEntrySelected(ComboBoxes_ViewModelMyOptionsWithFixedChoicesOption entryOption)
  {
  }
  void ComboBoxes_ViewModel::myOptionsWithDisplayValueComboBoxEntrySelected(std::string entryName)
  {
  }
  void ComboBoxes_ViewModel::myOptionsWithFixedChoicesAndDisplayValueComboBoxEntrySelected(ComboBoxes_ViewModelMyOptionsWithFixedChoicesAndDisplayValueOption entryOption)
  {
  }
}
